packages/libs/error-stack/src/fmt/color.rs

```diff
@@ -0,0 +1,161 @@
+use crate::{
+    fmt::r#override::{AtomicOverride, AtomicPreference},
+    Report,
+};
+
+/// The available modes of color support
+///
+/// Can be accessed through [`crate::fmt::HookContext::color_mode`], and set via
+/// [`Report::set_color_mode`]
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum ColorMode {
+    /// User preference to disable all colors
+    // TODO: this is only true once https://github.com/jam1garner/owo-colors/pull/90 is merged
+    // If this is the variant is present, [`owo-colors`](https://docs.rs/owo-colors) color
+    // support has been temporarily disabled and closures given to
+    // [`OwoColorize::if_supports_color`] will not be executed.
+    None,
+
+    /// User preference to enable colors
+    // TODO: this is only true once https://github.com/jam1garner/owo-colors/pull/90 is merged
+    // This will also temporarily set [`owo_colors::set_override`] to enable execution of
+    // [`OwoColorize::if_supports_color`].
+    Color,
+
+    /// User preference to enable styles, but discourage colors
+    ///
+    /// This is the same as [`ColorMode::Color`], but signals to the user that while colors are
+    /// supported, the user prefers instead the use of emphasis, like bold and italic text.
+    Emphasis,
+}
+
+impl ColorMode {
+    pub(super) fn load() -> Self {
+        COLOR_OVERRIDE.load()
+    }
+}
+
+impl Default for ColorMode {
+    #[cfg(feature = "color")]
+    fn default() -> Self {
+        Self::Emphasis
+    }
+
+    #[cfg(not(feature = "color"))]
+    fn default() -> Self {
+        Self::None
+    }
+}
```

Comment:
This feels wrong, especially when in mind, that we are going to remove the `color` feature in a follow-up. Can we just derive `Default` and set it to `Emphasis`?

Reply:
fixed in [a727237](https://github.com/hashintel/hash/pull/1800/commits/a727237a55d6a1fdcad4abc805e546800833dd14)